lib/http_requests.py

```diff
@@ -74,28 +74,84 @@ def get_http_data(url):
     return yml_data
 
 
-def get_advisories(branch_name):
-    url = f"{os.environ['GITHUB_RAW_CONTENT_URL']}/{branch_name}/releases.yml"
+def get_particular_advisory(data):
+    """
+    Function to get the advisories from a particular z-stream release
+    :data: The data of a particular version
+    """
+    try:
+        advisories = data['assembly']['group']['advisories']
+        if -1 in advisories.values() or 1 in advisories.values():
+            return None
+    except Exception:
+        return None
 
+    return advisories
+
+
+def get_brew_event_id(data):
+    """
+    Function to get the brew_event id from the releases.yml file
+    :data: The yml for a particular z-stream release
+    """
     try:
-        yml_data = get_http_data(url)['releases']
+        brew_event_id = data['assembly']['basis']['brew_event']
+        return brew_event_id
+    except KeyError:
+        return None
 
-        advisory_data = []
-        for version in yml_data:
-            try:
-                advisories = yml_data[version]['assembly']['group']['advisories']
-                brew_event = yml_data[version]['assembly']['basis']['brew_event']
-            except:
-                continue
 
-            if -1 in advisories.values() or 1 in advisories.values():
-                continue
-            advisory_data.append([brew_event, advisories])
+def get_advisories(branch_name):
+    url = f"{os.environ['GITHUB_RAW_CONTENT_URL']}/{branch_name}/releases.yml"
+    seen = {}  # variable to keep track of already processed openshift versions.
+    yml_data = get_http_data(url).get('releases', None)
 
-        return sorted(advisory_data, key=lambda x: x[0], reverse=True)
-    except Exception:
+    if not yml_data:
         return None
 
+    advisory_data = []
+    index = 0  # variable to sort by latest versions
+    for version in yml_data:
+        if version in seen:
+            continue  # If we have already included the advisories of a version
+
+        try:
+            _ = yml_data[version]['assembly']['basis']
+        except KeyError:
+            continue  # To make sure assembly and basis keys are present. Eg art3171 in 4.7 does not have those.
+
+        brew_event_id = get_brew_event_id(yml_data[version])
+
+        flag = False  # flag to check if a child that inherits has advisories, in case that has precedence
+        counter = 0  # counter to prevent infinite loops
```

Comment:
name these meaningfully. e.g.
```suggestion
        has_advisories = False  # flag to check if a child that inherits has advisories, in case that has precedence
        recursion_depth = 0  # counter to prevent infinite loops
```

Reply:
Tried to roll out the fix in a hurry, my bad.